google-ads-examples/build.gradle

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * This is the build file for the google-ads-examples subproject. It depends on
+ * the google-ads subproject.
+ *
+ * This project contains examples for using the Google Ads API. You can launch
+ * examples by navigating to the google-ads-examples folder and running the
+ * provided runExample gradle task. Note that we recommend running with the
+ * quiet flag (-q) to reduce screen clutter:
+ *     -- Mac/Linux --
+ *     ./gradlew -q runExample --example='basicoperations.GetCampaigns --customerId 1234567890'
+ *
+ *     -- Windows --
+ *     gradlew -q runExample --example='basicoperations.GetCampaigns --customerId 1234567890'
+ */
+
+plugins {
+    id 'com.google.api-ads.java-conventions'
+}
+
+dependencies {
+    implementation project(':google-ads')
+    implementation 'org.apache.commons:commons-lang3:3.11'
+    implementation 'com.beust:jcommander:1.72'
+    implementation 'joda-time:joda-time:2.8.2'
+    runtimeOnly 'org.apache.logging.log4j:log4j-slf4j-impl:2.11.1'
+    testImplementation 'org.mockito:mockito-core:2.27.0'
+}
+
+description = 'Google Ads API client library for Java examples'
+
+public class ExampleRunnerTask extends JavaExec {
```

Comment:
Is this class only needed to allow the user to specify without the `com.google....` prefix? If so is there perhaps an easier way to do this? The gradle build files are so pretty, it's a shame to put java boilerplate in them ;-)

Reply:
This seemed the best way to address the fact that this project has multiple "Main" classes.

The better, more Gradle-y way would be to have a single main class and to build the subproject as a Java application. Then you could just call `./gradlew run --args...`. This would, of course, involve some refactoring of the project.

Reply:
Could we move this to the buildSrc directory to simplify the build file here?

Reply:
Doing so would then apply the task to all subprojects, i.e. calling `./gradlew runExample ...` would run this code for the annotation processor, library, examples, and migration examples subprojects. You _could_ instead call `./gradlew google-ads-examples:runExample ...`, but that seems like an extra inconvenience when the task only applies to the one subproject.

Reply:
See related response to Josh's question below.